tests/integration_tests/test_precompiles.py

```diff
@@ -0,0 +1,156 @@
+import re
+
+import pytest
+
+from .ibc_utils import EVMOS_IBC_DENOM, assert_ready, get_balance, prepare_network
+from .utils import ADDRS, get_precompile_contract, wait_for_fn
+
+
+@pytest.fixture(scope="module", params=[False])
+def ibc(request, tmp_path_factory):
+    "prepare-network"
+    incentivized = request.param
+    name = "ibc-precompile"
+    path = tmp_path_factory.mktemp(name)
+    network = prepare_network(path, name, incentivized)
+    yield from network
+
+
+def test_ibc_transfer(ibc):
+    """
+    test transfer IBC precompile.
+    """
+    assert_ready(ibc)
+
+    dst_addr = ibc.chainmain.cosmos_cli().address("signer2")
+    amt = 1000000
+
+    cli = ibc.evmos.cosmos_cli()
+    src_addr = cli.address("signer2")
+    src_denom = "aevmos"
+
+    old_src_balance = get_balance(ibc.evmos, src_addr, src_denom)
+    old_dst_balance = get_balance(ibc.chainmain, dst_addr, EVMOS_IBC_DENOM)
+
+    pc = get_precompile_contract(ibc.evmos.w3, "ICS20I")
+    evmos_gas_price = ibc.evmos.w3.eth.gas_price
+
+    tx_hash = pc.functions.transfer(
+        "transfer",
+        "channel-0",
+        src_denom,
+        amt,
+        ADDRS["signer2"],
+        dst_addr,
+        [1, 10000000000],
+        0,
+        "",
+    ).transact({"from": ADDRS["signer2"], "gasPrice": evmos_gas_price})
+
+    receipt = ibc.evmos.w3.eth.wait_for_transaction_receipt(tx_hash)
+
+    assert receipt.status == 1
+    # check gas used
+    assert receipt.gasUsed == 127581
+
+    fee = receipt.gasUsed * evmos_gas_price
+
+    new_dst_balance = 0
+
+    def check_balance_change():
+        nonlocal new_dst_balance
+        new_dst_balance = get_balance(ibc.chainmain, dst_addr, EVMOS_IBC_DENOM)
+        return old_dst_balance != new_dst_balance
+
+    wait_for_fn("balance change", check_balance_change)
+    assert old_dst_balance + amt == new_dst_balance
+    new_src_balance = get_balance(ibc.evmos, src_addr, src_denom)
+    assert old_src_balance - amt - fee == new_src_balance
+
+
+def test_ibc_transfer_invalid_packet(ibc):
+    """
+    test transfer IBC precompile invalid packet error.
+    NOTE: it is important for this error message to not change
+    because it is already stored on mainnet.
+    Changing this error message is a state breaking change
+    """
+    assert_ready(ibc)
+
+    # IMPORTANT: THIS ERROR MSG SHOULD NEVER CHANGE OR WILL BE A STATE BREAKING CHANGE ON MAINNET
+    exp_err = "constructed packet failed basic validation: packet timeout height and packet timeout timestamp cannot both be 0: invalid packet"  # noqa: E501
```

Comment:
what is the `# noqa: E501` for? I'm not a fan of linter escapes without a reason

Reply:
probably, I would still add this explicitly as a comment there too 👍

Reply:
yes, it is line length lint. The comment is to mute that linter error